src/NUnitFramework/framework/Constraints/DictionaryContainsKeyValuePairConstraint.cs

```diff
@@ -70,6 +71,33 @@ private bool Matches(object? actual)
                 return false;
             }
 
+            // If 'actual' implements IEnumerable<KeyValuePair<TKey, TValue>>, we can find the value using that interface
+            var enumerableType = actual.GetType().GetInterface("IEnumerable`1");
```

Comment:
Suggest to cache the result of `actual.GetType().GetInterfaces()` from line 60.

Reply:
Not relevant if the original test on line 60 is changed to test for `IEnumerable<KeyValuePair<,>>` instead.